File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/ecs/component/Mana.java
package net.bestia.zoneserver.ecs.component;

import com.artemis.Component;

/**
 * Adds a regenerating mana pool to a entity. The current mana and max mana must be synced with the entities in the
 * background.
 * 
 * @author Thomas Felix <thomas.felix@tfelix.de>
 *
 */
public class Mana extends Component {

	public int maxMana;
	public int curMana;
	
}

File: zone-server/src/main/kotlin/net/bestia/zone/ecs/battle/Mana.kt
package net.bestia.zone.ecs.battle

import net.bestia.zone.ecs.Component
import net.bestia.zone.status.CurMax

class Mana(
  current: Int,
  max: Int
) : CurMax(), Component {

  init {
    this.max = max
    this.current = current
  }
}